akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala

```diff
@@ -514,44 +523,56 @@ private[persistence] trait Eventsourced extends Snapshotter with PersistenceStas
    *
    * All incoming messages are stashed.
    */
-  private def recovering(recoveryBehavior: Receive, previousReceiveTimeout: Duration) =
+  private def recovering(recoveryBehavior: Receive) =
     new State {
+
+      // protect against snapshot stalling forever because of journal overloaded and such
+      val timeout = extension.journalConfigFor(journalPluginId).getMillisDuration("recovery-event-timeout")
```

Comment:
Could I even put it in a place so it doesn't have to be read once per persistent actor start?